### PR TITLE
Update initialize-database.rst

### PR DESCRIPTION
**Pull request by jmfreeman**
Remove ambiguity in the language as only MySQL and Postgres are supported backends.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by XD-DENG — CHANGES_REQUESTED** [review]
SQLite should be mentioned given it's important for development/testing purpose.

Almost all new users start with SQLite (e.g. to test if the installation/setting-up is working). It will be weird to exclude it here.

Edit: what I meant is to *mention* SQLite (and highlight it's primarily for development purpose), to ensure consistent information. Otherwise people having quick looking at this may have the confusion "SQLite is not supported".

File: docs/howto/initialize-database.rst [hunk 1]
```diff
@@ -23,9 +23,8 @@ Initializing a Database Backend
 If you want to take a real test drive of Airflow, you should consider
 setting up a real database backend and switching to the LocalExecutor.
 
-As Airflow was built to interact with its metadata using the great SqlAlchemy
-library, you should be able to use any database backend supported as a
-SqlAlchemy backend. We recommend using **MySQL** or **Postgres**.
+Airflow was built to interact with its metadata using SqlAlchemy
+with **MySQL** and **Postgres** as supported backends.
```
Comment (potiuk) [inline docs/howto/initialize-database.rst]:
```suggestion
with **MySQL**,  **Postgres** and **SQLite** as supported backends (SQLite is used primarily for development purpose).
```

**Review by potiuk — APPROVED** [review]
Approved with the "SQLite" modification

File: docs/howto/initialize-database.rst (hunk 1 above)
Comment (XD-DENG) [reply docs/howto/initialize-database.rst]: maybe not making perfect sense to me, correct me if I'm wrong @potiuk :

This page is meant for users who have already installed Airflow and tried with SQLite (the earlier paragraph is `If you want to take a real test drive of Airflow, you should consider setting up a real database backend and switching to the LocalExecutor.`)

So I would like to suggest to apply your change + append with "`(SQLite is used primarily for development purpose)`". In addition I think you missed the "`as`" in your change ;-)

Sorry for being quite picky for documentation. It's quite essential to me though.

Comment (potiuk) [reply docs/howto/initialize-database.rst]: Well. There are anecdotal evidences of people using SQLite and sequential executor in production :). 
And you know @XD-DENG, And if you look closely, SQLite is one of the very few "True Serverless" databases https://sqlite.org/serverless.html , so in the  "Serverless" hype maybe everyone should switch to it ;) 

But on a more serious note - yeah. I think adding a note is a good idea  :)

Comment (XD-DENG) [reply docs/howto/initialize-database.rst]: I do like SQLIte as well.

The sentence I suggested to add is actually copied from current version of README. So it should be consistent

**Comment by XD-DENG** [comment]
@potiuk I made some minor changes on top of your suggestion (add "as", and fix the trailing comma to period.)

**Review by XD-DENG — APPROVED** [review]

**Comment by XD-DENG** [comment]
Thanks @jmfreeman for the contribution. Hope you are ok with the changes that @potiuk and I have made.

**Comment by potiuk** [comment]
Seems like one-sentence changes are those that gets the most scrutiny :) cool stuff!

**Comment by potiuk** [comment]
And we have a new contributor ;) @jmfreeman ... Looking forward to more contributions !

**Pull request merged**